Hive CI: step_entry_0 not found

Based on https://github.com/sql-machine-learning/sqlflow/pull/2970 .

```
2020-09-30T01:39:56.8637216Z [0m[0m=== RUN   TestRunSQLProgram
2020-09-30T01:40:10.6884767Z [0m[91m--- FAIL: TestRunSQLProgram (13.83s)
2020-09-30T01:40:10.6886874Z [0m[0m    executor_ir_test.go:143: 
2020-09-30T01:40:10.6888761Z [0m[0m        	Error Trace:	executor_ir_test.go:143
2020-09-30T01:40:10.6890171Z [0m[0m        	            				executor_ir_test.go:123
2020-09-30T01:40:10.6893117Z [0m[0m        	Error:      	Should be true
2020-09-30T01:40:10.6894200Z [0m[0m        	Test:       	TestRunSQLProgram
2020-09-30T01:40:10.6895524Z [0m[0m        	Messages:   	runSQLProgram error: exit status 1: Traceback (most recent call last):
2020-09-30T01:40:10.6897035Z [0m[0m        	            	  File "/tmp/sqlflow-couler-tmp805288536", line 5, in <module>
2020-09-30T01:40:10.6898446Z [0m[0m        	            	    print(pyfunc.body(step_entry_0))
2020-09-30T01:40:10.6899608Z [0m[0m        	            	NameError: name 'step_entry_0' is not defined
2020-09-30T01:40:10.6900669Z [0m[0m        	            	
2020-09-30T01:40:10.6902936Z [0m[0m        	            	Code is:
2020-09-30T01:40:10.6903776Z [0m[0m        	            	
2020-09-30T01:40:10.6904413Z [0m[0m        	            	
2020-09-30T01:40:10.6904961Z [0m[0m        	            	
2020-09-30T01:40:10.6905668Z [0m[0m        	            	import couler.pyfunc as pyfunc
2020-09-30T01:40:10.6906628Z [0m[0m        	            	print(pyfunc.body(step_entry_0))
```

--- a/python/runtime/model/db.py
+++ b/python/runtime/model/db.py
@@ -53,6 +53,8 @@ class SQLFSWriter(object):
 
     def write(self, content):
         block = base64.b64encode(content)
+        if six.PY3 and isinstance(block, bytes):
+            block = block.decode("utf-8")
         self.writer.write([self.row_idx, block])
         self.row_idx += 1
 
@@ -66,10 +68,27 @@ class SQLFSWriter(object):
         self.context_manager.__exit__(*args, **kwargs)
 
 
+def _build_ordered_reader(reader):
+    block_dict = dict()
+    cur_id = 0
+    for id, block in reader:
+        block_dict[id] = block
+        while True:
+            next_block = block_dict.pop(cur_id, None)
+            if next_block is None:
+                break
+
+            yield cur_id, next_block
+            cur_id += 1
+
+    assert not block_dict, "invalid model db format"
+
+
 class SQLFSReader(object):
     def __init__(self, conn, table):
-        sql = "SELECT block FROM {0} ORDER BY id".format(table)
-        self.rs = conn.query(sql)
+        sql = "SELECT id, block FROM {};".format(table)
+        self.raw_rs = conn.query(sql)
+        self.rs = _build_ordered_reader(self.raw_rs)
         self.reader = iter(self.rs)
         self.buffer = b''
 
@@ -85,7 +104,7 @@ class SQLFSReader(object):
             if new_buffer is None:
                 break
 
-            new_buffer = base64.b64decode(new_buffer[0])
+            new_buffer = base64.b64decode(new_buffer[1])
             self.buffer += new_buffer
 
         read_length = min(n, len(self.buffer))
@@ -94,7 +113,7 @@ class SQLFSReader(object):
         return result
 
     def close(self):
-        self.rs.close()
+        self.raw_rs.close()
 
     def __enter__(self, *args, **kwargs):
         return self

